src/ray/gcs/redis_context.cc

```diff
@@ -431,6 +431,49 @@ void ValidateRedisDB(RedisContext &context) {
   }
 }
 
+void ValidateAndConnectRedisSentinel(RedisContext &context, 
+                          const std::string &password,
+                          bool enable_ssl) {
+  auto reply = context.RunArgvSync(std::vector<std::string>{"INFO", "SENTINEL"});
+  if (reply->IsNil() || reply->IsError() || reply->ReadAsString().length() == 0) {
+    RAY_LOG(INFO) << "failed to get redis sentinel info, continue as a regular redis.";
+    return;
+  }
+  std::vector<const char *> argv;
+  std::vector<size_t> argc;
+  std::vector<std::string> cmds = {"SENTINEL", "MASTERS"};
+  for (const auto &arg : cmds) {
+    argv.push_back(arg.data());
+    argc.push_back(arg.size());
+  }
+
+  auto redis_reply = reinterpret_cast<redisReply *>(
+      ::redisCommandArgv(context.sync_context(), cmds.size(), argv.data(), argc.data()));
+
+  RAY_CHECK(redis_reply && redis_reply->type == REDIS_REPLY_ARRAY) << "failed to get redis sentinel masters info";
+  RAY_CHECK(redis_reply->elements == 1) << "expecting only one primary behind the redis sentinel";
+  auto primary = redis_reply->element[0];
+  std::string actual_ip, actual_port;
+  for (size_t i = 0; i < primary->elements; i += 2) {
+    std::string key = primary->element[i]->str;        // Key (e.g., "name", "ip")
+    std::string value = primary->element[i + 1]->str;  // Value corresponding to the key
+    if ("ip" == key) {
+      actual_ip = value;
+    } else if ("port" == key) {
+      actual_port = value;
+    }
+  }
+  freeReplyObject(redis_reply);
+  if (actual_ip.empty() || actual_port.empty()) {
+    RAY_LOG(FATAL) << "failed to get the ip and port of the master from redis sentinel";
+  } else {
+    RAY_LOG(INFO) << "fonnecting to the redis primary node behind sentinel: " << actual_ip << ":" << actual_port;
+    context.Disconnect();
+    Status status = context.Connect(actual_ip, std::stoi(actual_port), password, enable_ssl);
```

Comment:
(?) What happens then if the primary node is unhealthy there ? Would it just crash ?

Reply:
yes. that's similar behavior to current ray-redis setup: if ray can't talk to redis at any time, it will crash. 
I think it's a reasonable trade off (make service behavior clear and don't try to guess what would work if the current configuration is not working.

Reply:
Yeah that works for me ! I was curious, but yeah I think we can draw the reliability limit here :D